src/Apps/Order/OrderApp.tsx

```diff
@@ -8,13 +16,38 @@ export interface OrderAppProps {
     orderID: string
   }
   location: any
+  routeIndices: number[]
+  routes: any
 }
 
-export const OrderApp: SFC<OrderAppProps> = ({
-  me,
-  children,
-  location,
-  ...props
-}) => {
-  return <>{children}</>
+export class OrderApp extends React.Component<OrderAppProps> {
+  removeTransitionHook: () => void
+
+  constructor(props) {
+    super(props)
+
+    this.removeTransitionHook = props.router.addTransitionHook(
+      this.onTransition
+    )
+  }
+
+  componentWillUnmount() {
+    this.removeTransitionHook()
+  }
+
+  onTransition = location => {
+    const { routes, routeIndices } = this.props
+    let route = findRoute(routes, routeIndices)
+
+    if (route.onTransition) {
+      return route.onTransition(location)
```

Comment:
A transition will only be called for a route that has specified an `onTransition` parameter.

Reply:
I like this pattern 👍